functions-framework-api/src/main/java/com/google/cloud/functions/ExperimentalCloudEventsFunction.java

```diff
@@ -0,0 +1,22 @@
+package com.google.cloud.functions;
+
+import io.cloudevents.CloudEvent;
+
+/**
+ * Represents a Cloud Function that is activated by an event and parsed into a {@link CloudEvent} object.
+ * Because the {@link CloudEvent} API is not yet stable, a function implemented using this class may not
+ * build or work correctly with later versions of that API. Once the API is stable, this interface will
+ * become {@code CloudEventsFunction} and will also be stable.
+ */
+@FunctionalInterface
+public interface ExperimentalCloudEventsFunction {
+  /**
+   * Called to service an incoming event. This interface is implemented by user code to
+   * provide the action for a given background function. If this method throws any exception
+   * (including any {@link Error}) then the HTTP response will have a 500 status code.
```

Comment:
As an FYI, I've been thinking about whether it would be worth having a specific exception meaning "the function failed, but please use status code X".
See https://github.com/GoogleCloudPlatform/functions-framework-dotnet/pull/61
Thoughts welcome. (We'd want to do this across languages, I suspect.)

Reply:
Instead of always throwing 500s, I would assume we could throw a 400 bad request for a malformed CE and other 4XX errors implying a user error.

https://github.com/GoogleCloudPlatform/functions-framework#http-status-codes

Reply:
It depends on what detects that the CloudEvent is "malformed" - if it's "malformed" in that the user code can't deserialize the data, for example, then we need a way of communicating that.
If it's malformed in terms of (say) not having an event ID, then I'd expect that to mean the user's code is never run, in which case this paragraph is irrelevant (as it only talks about what happens when the user code throws an exception).

Reply:
I do like this idea, but I don't know how well it will map to all of our languages.